tests/DecodeChunkedStreamTest.php

```diff
@@ -0,0 +1,143 @@
+<?php
+
+namespace React\Tests\HttpClient;
+
+use Exception;
+use React\HttpClient\ChunkedStreamDecoder;
+use React\Stream\ThroughStream;
+
+class DecodeChunkedStreamTest extends TestCase
+{
+    public function provideChunkedEncoding()
+    {
+        return [
+            [
```

Comment:
Perhaps we should assign names to the data sets? The outer array *can* use assoc keys to describe what each data set does and eventually may ease debugging. (see also below)